src/Arcus.Messaging.Tests.Unit/ServiceBus/ObjectExtensionsTests.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.ServiceBus.Core.Extensions;
+using Arcus.Messaging.Tests.Core;
+using Arcus.Messaging.Tests.Core.Generators;
+using Arcus.Messaging.Tests.Core.Messages.v1;
+using Microsoft.Azure.ServiceBus;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Arcus.Messaging.Tests.Unit.ServiceBus
+{
+    [Trait("Category", "Unit")]
+    public class ObjectExtensionsTests
+    {
+        private const string ExpectedDefaultEncoding = "utf-8";
+        private const string ExpectedDefaultContentType = "application/json";
+
+        [Fact]
+        public void WrapInServiceBusMessage_BasicWithoutOptions_ReturnsValidServiceBusMessage()
+        {
+            // Arrange
+            var messagePayload = OrderGenerator.Generate();
+
+            // Act
+            var serviceBusMessage = messagePayload.WrapInServiceBusMessage();
+
+            // Assert
+            Assert.NotNull(serviceBusMessage);
+            Assert.Null(serviceBusMessage.CorrelationId);
+            var userProperties = serviceBusMessage.UserProperties;
```

Comment:
This will probably feel again like nitpicking.
A lot of assert lines are made for the `userProperties` with rather "basic" assertions. (IsTrue, IsFalse ...). xUnit has several collection asserts (`All`, `Contains`, `Collection`, ...) that maybe make these somewhat cleaner and more discoverable when something is wrong.
Just commenting, not blocking.

Reply:
Would you mind giving a more detailed example please as I'm not seeing how I could change it.

Instead of `Assert.True(userProperties.ContainsKey(PropertyNames.ContentType));` I can use `Contains` indeed but it would look like `Assert.Contains(userProperties, (pair) => pair.Key.Equals(PropertyNames.ContentType, StringComparison.InvariantCultureIgnoreCase));`

Reply:
This:

```csharp
 var userProperties = serviceBusMessage.UserProperties;
 Assert.True(userProperties.ContainsKey(PropertyNames.ContentType));
 Assert.True(userProperties.ContainsKey(PropertyNames.Encoding));
 Assert.False(userProperties.ContainsKey(PropertyNames.TransactionId));
 ArcusAssert.MatchesDictionaryEntry(PropertyNames.ContentType, ExpectedDefaultContentType, userProperties);
 ArcusAssert.MatchesDictionaryEntry(PropertyNames.Encoding, ExpectedDefaultEncoding, userProperties);
 AssertMessagePayload(serviceBusMessage, userProperties, ExpectedDefaultEncoding, messagePayload);
```

Is the same as:

```csharp
Assert.Collection(
    userProperties,
    kv => Assert.Equal(kv, new KeyValuePair<string, object>(PropertyNames.ContentType, ExpectedDefaultContentType)),
    kv =>
    {
        Assert.Equal(kv, new KeyValuePair<string, object>(PropertyNames.Encoding, ExpectedDefaultEncoding));
        AssertMessagePayload(serviceBusMessage, messagePayload, kv);
    });
```

Reply:
That syntax is not really clear imo but ok, I'll see if I can make it change.

Reply:
I've decided not to go this route as `Assert.Collection` assumes the order of the list is the same as our asserts.

In other words, if we change the extension and add Encoding before ContentType, then the test will break while it's still valid which is not what we want.

Reply:
Ok, I'll leave it up to you.